internal/proxy/fuse_test.go

```diff
@@ -43,21 +44,25 @@ func randTmpDir(t interface {
 // newTestClient is a convenience function for testing that creates a
 // proxy.Client and starts it. The returned cleanup function is also a
 // convenience. Callers may choose to ignore it and manually close the client.
-func newTestClient(t *testing.T, d cloudsql.Dialer, fuseDir, fuseTempDir string) (*proxy.Client, func()) {
+func newTestClient(t *testing.T, d cloudsql.Dialer, fuseDir, fuseTempDir string) (*proxy.Client, chan error, func()) {
 	conf := &proxy.Config{FUSEDir: fuseDir, FUSETempDir: fuseTempDir}
 	c, err := proxy.NewClient(context.Background(), d, testLogger, conf)
 	if err != nil {
 		t.Fatalf("want error = nil, got = %v", err)
 	}
 
 	ready := make(chan struct{})
-	go c.Serve(context.Background(), func() { close(ready) })
+	servErrCh := make(chan error)
+	go func() {
+		servErr := c.Serve(context.Background(), func() { close(ready) })
+		servErrCh <- servErr
```

Comment:
This will leak goroutines if servErrCh is full.

How about:

``` golang
select {
  case servErrCh <- servErr:
  default:
    // exit background thread
}
```

Reply:
Fixed.